Search directory from GOROOT/src and modules

qingshan opened the pull request:
If The -dir option is relative, will try to find the absolute directory from `GOROOT/src` and modules. This behavior should be same as `go doc <pkg>`. This code is copied from `GOROOT/src/cmd/doc` with minor changes. changes summary here:

If -dir option is:
- absolute path(starts with "/"), no change
- relative path (starts with "./" or "../"), apply with current work directory.
- otherwise,  search as a suffix of a path from  `GOROOT/src` and modules.
- if nothing found, use -dir option anyway.

fatih commented:
Thank you @qingshan 

`motion` only works with absolute paths. I don't have any plan to change the behavior it as it works today with no issues. If you need that feature, feel free to fork and compile it yourself.

qingshan commented:
Thank you for the update!, I extracted the code outside of `motion`, and implemented a find tool to search directory and pass to motion. so that could keep `motion` unchanging.
